✨(apps) allow defining resources requests for all pods

madmatah opened the pull request:
## Purpose
We need to improve pod scheduling on our Kubernetes clusters.

This PR is based on the work done in #353 

## Proposal
By setting resources requests without any limit, our pods are in
"Burstable" mode: we can define the minimum resources it requires upon
scheduling, while still allowing it to burst.

- [x] Rebase #353 with `migration-to-k8s`
- [x] Allow to use jinja in app defaults files
- [x]  Add missing apps
  - [x] ashley
  - [x] edxec
  - [x] elasticsearch
  - [x] etherpad
  - [x] flower
  - [x] kibana
  - [x] redis-sentinel
- [x] Add limits in cronjobs
- [x] Set default limits that reflect real usage.

sampaccoud committed:
✨(apps) allow defining resources requests for all pods

We need to improve pod scheduling on our Kubernetes clusters. By
setting resources requests without any limit, our pods are in
"Burstable" mode: we can define the minimum resources it requires upon
scheduling, while still allowing it to burst.

sampaccoud commented on group_vars/customer/eugene/main.yml:
```diff
@@ -13,6 +13,10 @@ nextcloud_database_name: nextcloud
 elasticsearch_java_opts:
   - "-Xms128m"
   - "-Xmx128m"
+elasticsearch_app_resources:
```
I didn't understand why Eugene needs to override this variable?

madmatah replied:
Because 512MB (default value) is too much for a dev CI/env, it has been configured to use 128MB instead.

sampaccoud replied:
I get the idea: Eugene is only for dev/CI so let's limit it.

madmatah committed:
✨(core) allow to use jinja in apps defaults templates

We want be able to declare reusable variables in app defaults to avoid
code duplication.

jmaupetit commented on tasks/load_app_defaults.yml:
```diff
@@ -1,9 +1,8 @@
-    file: "{{ file }}"
-    name: defaults
+  set_fact:
+    defaults: "{{ lookup('template', file) | from_yaml }}"
```
![](https://media2.giphy.com/media/5FoxedYWNVmO4/giphy.gif)

jmaupetit reviewed (approved):
![](https://media2.giphy.com/media/3ohhwj5bIw0EydVp3q/giphy.gif)

madmatah commented:
Okay everything seems to be fine.
I adjusted resources request values based on running application on a production environment.
It might have to be adjusted again because I didn't have historical data, but it is a good start.
